tests/unit/utils/test_sqlalchemy_exec_utils.py

```diff
@@ -84,18 +92,22 @@ def error_during_iteration():
 
     # list() is always called on the successful return value
     assert execute_with_retry() == ["it's", "OK", "now"]
-    assert execute.call_count == 4
+    assert connection.execute.call_count == 4
     assert sleep.mock_calls == [mock.call(t) for t in [10, 20, 40]]
     assert "Retrying query (attempt 3 / 3)" in log_messages
 
 
 @mock.patch("time.sleep")
 def test_execute_with_retry_exhausted(sleep):
-    execute = mock.Mock(side_effect=[ERROR, ERROR, ERROR, ERROR])
+    connection = mock.Mock(
+        **{
```

Comment:
TIL about this slightly obtuse way of using dotted notation to define nested attributes